metricbeat/helper/prometheus/textparse.go

```diff
@@ -577,49 +586,67 @@ func ParseMetricFamilies(b []byte, contentType string, ts time.Time) ([]*MetricF
 		var metric *OpenMetric
 
 		metricName := lset.Get(labels.MetricName)
-		var lookupMetricName string
+
+		// lookupMetricName will have the suffixes removed
+		lookupMetricName := metricName
 		var exm *exemplar.Exemplar
 
 		// Suffixes - https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#suffixes
 		switch mt {
 		case textparse.MetricTypeCounter:
+			if contentType == OpenMetricsType && !isTotal(metricName) && !isCreated(metricName) {
+				// Possible suffixes for counter in Open metrics are _created and _total.
+				// Otherwise, ignore.
+				// https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#counter-1
+				continue
+			}
+
 			var counter = &Counter{Value: &v}
 			mn := lset.Get(labels.MetricName)
 			metric = &OpenMetric{Name: &mn, Counter: counter, Label: labelPairs}
-			if isTotal(metricName) && contentType == OpenMetricsType { // Remove suffix _total, get lookup metricname
-				lookupMetricName = strings.TrimSuffix(metricName, suffixTotal)
+			if contentType == OpenMetricsType {
+				// Remove the two possible suffixes, _created and _total
+				if isTotal(metricName) {
+					lookupMetricName = strings.TrimSuffix(metricName, suffixTotal)
+				} else {
```

Comment:
```suggestion
				} isCreated(metricName) {
```

Reply:
Because of this condition above `if contentType == OpenMetricsType && !isTotal(metricName) && !isCreated(metricName)` we already know that if we are in the `else`, the suffix is `_created` @gizas